tools/firmware/build_firmware.sh

```diff
@@ -5,22 +5,29 @@ mkdir -p _release
 
 # Variables
 RECOMPILE_BOOTLOADER="false"
-while getopts r flag
+while getopts rv: flag
 do
     case "${flag}" in
         r) RECOMPILE_BOOTLOADER="true";;
+        v) APPLICATION_VERSION=$OPTARG;;
     esac
 done
 
 BOOTLOADER_HEX="_tmp/bootloader.hex"
 
 APPLICATION_HEX_SOURCE="_build/LEKA_V1_2_DEV/app/os/LekaOS.hex"
-APPLICATION_VERSION="1.1.0+0"
 APPLICATION_SIGNED_HEX="_tmp/application-signed.hex"
 
 FIRMWARE_HEX="_release/firmware.hex"
 FIRMWARE_RELEASE="_release/firmware.bin"
 
+APPLICATION_VERSION_REGEX="^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,5}(\+[0-9]+)?$"
+if [ -z "$APPLICATION_VERSION" ] || [[ ! $APPLICATION_VERSION =~ $APPLICATION_VERSION_REGEX ]];
+then
+    echo "Invalid version, exiting script..."
+    exit 0
```

Comment:
because it's a failure, the script must return 1, not 0

in shells, 0 == success, anything else == failure

```suggestion
    exit 1
```

Reply:
ok